### PR TITLE
Fix prepare_package.dart not updating base_url

### LINKED ISSUE #81870: releases json manifests "base_url" is old "flutter_infra" storage bucket
```json
  "base_url": "https://storage.googleapis.com/flutter_infra/releases",
  "current_release": {
    "beta": "b5017bf8de877083978bfeb1874d236c3fc83029",
...
```

Should instead point to the new `flutter_infra_release` bucket:

```json
  "base_url": "https://storage.googleapis.com/flutter_infra_release/releases",
  "current_release": {
    "beta": "b5017bf8de877083978bfeb1874d236c3fc83029",
...
```
https://storage.googleapis.com/flutter_infra_release/releases/releases_macos.json
https://storage.googleapis.com/flutter_infra_release/releases/releases_windows.json
https://storage.googleapis.com/flutter_infra_release/releases/releases_linux.json

### PR DESCRIPTION
**Pull request by christopherfujino**
The issue was that the parameter `newBucket` of the `_updateMetadata()` method was always being passed `false`, even for the new bucket manifest, and this value was used to fill in the `base_url` value.

Fixes https://github.com/flutter/flutter/issues/81870

### CODE REVIEW

File: dev/bots/test/prepare_package_test.dart [hunk 1]
```diff
@@ -408,6 +408,89 @@ void main() {
         expect(contents, equals(encoder.convert(jsonData)));
       });
 
+
```
Comment (jmagman) [inline dev/bots/test/prepare_package_test.dart:411]: Extra newline

**Review by jmagman — COMMENTED** [review]
LGTM

**Comment by christopherfujino** [comment]
> LGTM

Can you please approve :)

File: dev/bots/test/prepare_package_test.dart (hunk 1 above)
Comment (christopherfujino) [reply dev/bots/test/prepare_package_test.dart]: done

**Review by jmagman — APPROVED** [review]

**Pull request merged**